index.js

```diff
@@ -523,16 +523,28 @@ function transformAllModels(models) {
 /*
  * Extends an object with another
  * @param source {object} - object that will get extended
- * @parma distention {object} - object the will used to extend source
+ * @parma destination {object} - object the will used to extend source
 */
-function extend(source, distention) {
+function extend(source, destination) {
   if (typeof source !== 'object') {
     throw new Error('source must be objects');
   }
 
-  if (typeof distention === 'object') {
-    Object.keys(distention).forEach(function(key) {
-      source[key] = distention[key];
+  if (typeof destination === 'object') {
+    Object.keys(destination).forEach(function(key) {
+      source[key] = destination[key];
     });
   }
 }
+
+/*
+ * Convert string values into the proper type.
+ * @param value {*} - value to convert
+ * @returns {*} - transformed modles object
+*/
+function fixValueType(value) {
+  if (typeof value !== 'string') {
+    return value;
+  }
+  return JSON.parse(value);
```

Comment:
>Are you using this method?
Yes. Inside `processDataType` function
>This will always throw actually
Why? I pass string to JSON.parse, so it should throw only if string is invalid.
Meaning it not boolean, number, array, object, etc.
> Please guard against it.
I can just provide better error message, but I can't recover from such error.